salt/modules/yumpkg.py

```diff
@@ -245,14 +245,27 @@ def _get_branch_option(**kwargs):
     Returns a list of '--branch' option to be used in the yum command,
     based on the kwargs. This feature requires 'branch' plugin for YUM.
     '''
-    branch = kwargs.get('branch', '')
+    branch = kwargs.pop('branch', '')
     ret = []
     if branch:
         log.info('Adding branch \'%s\'', branch)
         ret.append('--branch=\'{0}\''.format(branch))
     return ret
 
 
+def _get_extra_options(**kwargs):
+    '''
+    Returns list of extra options for yum
+    '''
+    ret = []
+    for key, value in kwargs.items():
```

Comment:
We should also make sure that we are stripping the pub kwargs using ``salt.utils.clean_kwargs()``, otherwise we will be adding all of the pub kwargs to the command (e.g. ``--__pub_jid=01234...``) which will cause problems.